scripts/run-clang-format.sh

```diff
@@ -1,73 +1,53 @@
-#! /bin/bash
+#!/bin/bash
 #
-# run-clang-format.sh: formats source code in this repo in accordance with .clang-format file.
+# run-clang-format.sh: Formats source code in this repo in accordance with .clang-format file.
 # This file is part of the DP3 software package.
 # Copyright (C) 2020 ASTRON (Netherlands Institute for Radio Astronomy)
 # License: GNU General Public License version 3 or any later version
+#
+# To hook this script to pre-commit include the line
+# "./scripts/run-clang-format.sh" to .git/hooks/pre-commit
+# and make sure pre-commit is an executable shell script.
 
-programname=$0
+#Script configuration for this repo. Adjust it when copying to a different repo.
 
-set -e
+#The directories that contain source files, which clang-format should format.
+SOURCE_DIRS=($(dirname "$0")/..)
 
```

Comment:
I would opt for one `SOURCE_DIR` rather than providing the option to specify a list of `SOURCE_DIRS`. Since `find` will work recursively, there doesn't seem much of a use case for specifying a list of directories. 

What's more: if you have only one `SOURCE_DIR` it's convenient to `cd` into this directory before executing `find`. This allows to provide relative paths to the root of `SOURCE_DIR` in `EXCLUDE_DIRS`